Contrib/ChiralPairs/ChiralDescriptors.py

```diff
@@ -0,0 +1,300 @@
+                numNOs+=1.0/sharedNeighbors_dict[i]
+            if sharedNeighbors_dict[i] > 1:
+                numShared+=1
+        numRotBs = getRotatableBondsSubstituent(mol, set(sub))
+        aroBonds = getAromaticBondsSubstituent(mol, set(sub))
+        # fill the substituentDescriptor tuple
+        sizeDict[sidx]=substituentDescriptor(size=size,relSize=size/numAtoms,numNO=numNOs,relNumNO=numNOs/numAtoms,
+                                             relNumNO_2=numNOs/size,pathLength=maxShell_dict[sidx],
+                                             relPathLength=maxShell_dict[sidx]/numAtoms,relPathLength_2=maxShell_dict[sidx]/size,
+                                                 sharedNeighbors=numShared, numRotBonds=numRotBs, numAroBonds=aroBonds)
+    # if we have less then 4 substituents the missing ones need to be an hydrogen atoms
+    if len(sizeDict) < 4:
+        for i in range(4-len(sizeDict)):
+            sizeDict['H'+str(i)]=substituentDescriptor(size=0,relSize=0,numNO=0,relNumNO=0,relNumNO_2=0,
+                                                       pathLength=0,relPathLength=0,relPathLength_2=0,sharedNeighbors=0, numRotBonds=0,
+                                                       numAroBonds=0)
+    return sizeDict
+
+
+# Visualization of the substituents
+def visualizeSubstituentsGrid(mol, aIdx, molSize=(300,150), kekulize=True,):
+    dists = Chem.GetDistanceMatrix(mol)
+    idxChiral = Chem.FindMolChiralCenters(mol)[0][0]
+    subDict, sn_dict, maxShell_dict = determineAtomSubstituents(aIdx, mol, dists, False)
+    
+    colors = sns.husl_palette(len(subDict), s=.6)    
+    mc = rdMolDraw2D.PrepareMolForDrawing(mol, kekulize=kekulize)
+    count=0
+    svgs=[]
+    labels=[]
+    for sidx, sub in sorted(subDict.items(), key= lambda x: _getSizeOfSubstituents(x[1], sn_dict)):
+        color = tuple(colors[count])        
+        count+=1
+        atColors={}
+        for atom in sub:
+            atColors[atom]=color
+        
+        bonds = getBondsSubstituent(mol, set(sub)) 
+        bnColors={}
+        for b in bonds:
+            bnColors[b]=color
+        
+        drawer = rdMolDraw2D.MolDraw2DSVG(molSize[0],molSize[1])
+        drawer.DrawMolecule(mc,highlightAtoms=atColors.keys(),
+                            highlightAtomColors=atColors,highlightBonds=bonds,highlightBondColors=bnColors)
+        drawer.FinishDrawing()
+        svg = drawer.GetDrawingText()
+        svgs.append(svg.replace('svg:',''))
+        labels.append("Substituent "+str(count)+" (#atoms: "+str(len(sub))+", size normed: "+str(_getSizeOfSubstituents(sub, sn_dict))+")")
+    return _svgsToGrid(svgs, labels, svgsPerRow=len(svgs),molSize=molSize,fontSize=12)    
+    
+def visualizeChiralSubstituentsGrid(mol):
+    idxChiral = Chem.FindMolChiralCenters(mol)[0][0]
+    return visualizeSubstituentsGrid(mol, idxChiral)
+
+# Chiral moment descriptor    
+from numpy import linalg as LA
```

Comment:
import statements always at the top. I think you don't win anything from abbreviating linalg to LA. You only use it once anyway.